### PR TITLE
remove ps-node module (closes #2860)

### LINKED ISSUE #2860: v0.22.0 on Mac regression - 'browser not closing after test' is re-occurring
### Are you requesting a feature or reporting a bug?
bug

### What is the current behavior?
Looks like a regression of this issue: https://github.com/DevExpress/testcafe/issues/2132

In v0.22.0 - after test completes, Chrome process remains open (in dock)

### What is the expected behavior?
In v0.20.5 - after test completes, Chrome browser closes

### How would you reproduce the current behavior (if this is a bug)?
Run test using Chrome: 
```
npx testcafe chrome test.js
```

#### Provide the test code and the tested page URL (if applicable)

Tested page URL:
https://devexpress.github.io/testcafe/example
Test code

```js
// From your github page: https://github.com/DevExpress/testcafe
import { Selector } from 'testcafe'; // first import testcafe selectors

fixture `Getting Started`// declare the fixture
    .page `https://devexpress.github.io/testcafe/example`;  // specify the start page

//then create a test and place your code there
test('My first test', async t => {
    await t
        .typeText('#developer-name', 'John Smith')
        .click('#submit-button')

        // Use the assertion to check if the actual header text is equal to the expected one
        .expect(Selector('#article-header').innerText).eql('Thank you, John Smith!');
});
```

### Specify your

* operating system: Mac Sierra 10.12.6
* testcafe version: 0.22.0
* node.js version: 10.9.0

### PR DESCRIPTION
**Pull request by AlexKamaev**

### CODE REVIEW

File: src/utils/promisified-functions.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import childProcess from 'child_process';
 import fs from 'graceful-fs';
-import psNode from 'ps-node';
+import { lookup, kill } from './close-process';
```
Comment (AndreyBelym) [inline src/utils/promisified-functions.js]: I think it's better to name it just `process`.

File: src/utils/close-process.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    catch (e) {
+        next(e);
+    }
+
+    const killTimeoutTimer = setTimeout(() => {
+        throw new Error('Kill process timeout');
+    }, CHECK_PROCESS_IS_KILLED_TIMEOUT);
+
+    const checkKilled = () => {
+        getProcessById(pid, (err, isProcessNotKilled) => {
```
Comment (miherlosev) [inline src/utils/close-process.js]: I guess we should add a delay between `checkKilled`

File: src/utils/close-process.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    const processIdRegex   = new RegExp('^\\s*' + processId + '\\s+.*');
+    const lines            = psOutput.split(NEW_LINE_SEPERATOR_RE);
+
+    return lines.some(line => processIdRegex.test(line));
+};
+
+const getProcessOutput = callback => {
+    const child = childProcess.spawn('ps', ['-eo', 'pid,command']);
+    let stdout  = '';
+    let stderr  = null;
```
Comment (AndreyBelym) [inline src/utils/close-process.js]: You can use `stderr = ''` here, and you won't need the `if` operator in the `data` event handler.

File: src/utils/close-process.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
+import childProcess from 'child_process';
+
+const CHECK_PROCESS_IS_KILLED_TIMEOUT = 5000;
+const NEW_LINE_SEPERATOR_RE           = /(\r\n)|(\n\r)|\n|\r/g;
+
+const findProcessId = (browserId, psOutput) => {
```
Comment (AndreyBelym) [inline src/utils/close-process.js]: AFAIK we use `function` definitions for non-callback functions, don't we?

File: src/utils/kill-browser-process.js [hunk 5]
```diff
@@ -43,17 +40,25 @@ async function findProcessWin (processOptions) {
 }
 
 export default async function (browserId) {
-    const processOptions = { arguments: browserId, psargs: '-ef' };
-    const processList    = OS.win ? await findProcessWin(processOptions) : await findProcess(processOptions);
+    let processId = null;
+
+    if (OS.win) {
```
Comment (AndreyBelym) [inline src/utils/kill-browser-process.js]: Maybe we can move this condition and `findProcessWin` into the `./process` module?

File: src/utils/close-process.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+};
+
+const isProcessExist = (processId, psOutput) => {
+    const processIdRegex   = new RegExp('^\\s*' + processId + '\\s+.*');
+    const lines            = psOutput.split(NEW_LINE_SEPERATOR_RE);
+
+    return lines.some(line => processIdRegex.test(line));
+};
+
+const getProcessOutput = callback => {
```
Comment (AndreyBelym) [inline src/utils/close-process.js]: You should handle `error` event too. Sometimes `child_process` emits only `error` event, and no `exit`.

File: src/utils/promisified-functions.js [hunk 7]
```diff
@@ -10,8 +10,8 @@ export const writeFile  = promisify(fs.writeFile);
 export const readFile   = promisify(fs.readFile);
 export const deleteFile = promisify(fs.unlink);
 
-export const findProcess = promisify(psNode.lookup);
-export const killProcess = promisify(psNode.kill);
+export const findProcess = promisify(lookup);
```
Comment (AndreyBelym) [inline src/utils/promisified-functions.js]: Please, use Promises in the original functions, avoid using callbacks.

**Pull request merged**